### PR TITLE
Make Response[<T>] easier to use in debugger

### PR DESCRIPTION
**Pull request by pakrym**
Adds ToString and debugger proxy to show GetRawResponse value:
image

### CODE REVIEW

File: sdk/core/Azure.Core/src/Shared/NoBodyResponse{T}.cs [hunk 1]
```diff
@@ -24,6 +24,11 @@ public override T Value
 
         public override Response GetRawResponse() => _response;
 
+        public override string ToString()
+        {
+            return $"Status: {GetRawResponse().Status}, Service returned no content";
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Shared/NoBodyResponse{T}.cs]: Can you add similar to DelayedResponse in batching?  I think the base ToString might cause it to throw.

File: sdk/core/Azure.Core/src/Internal/ResponseDebugView.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    internal class ResponseDebugView<T>
+    {
+        private readonly Response<T> _response;
+
+        public ResponseDebugView(Response<T> response)
+        {
+            _response = response;
+        }
+
+        public Response GetRawResponse => _response.GetRawResponse();
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Internal/ResponseDebugView.cs]: Nice

**Review by tg-msft — APPROVED** [review]
Wish we could put `Value` over `GetRawResponse`, but this is a great change.

File: sdk/storage/Azure.Storage.Blobs.Batching/src/DelayedResponse.cs [hunk 3]
```diff
@@ -77,6 +77,11 @@ public void SetLiveResponse(Response live, bool throwOnFailure)
             }
         }
 
+        public override string ToString()
+        {
+            return _live == null ? "Status: NotSent, Request is not sent yet." : base.ToString();
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs.Batching/src/DelayedResponse.cs]: Nit - can we change this to `Status: NotSent, the batch has not been submitted yet.` to be super explicit if someone sees this while debugging?

**Review by tg-msft — APPROVED** [review]

File: sdk/core/Azure.Core/src/Shared/NoBodyResponse{T}.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Shared/NoBodyResponse{T}.cs]: DOne

File: sdk/storage/Azure.Storage.Blobs.Batching/src/DelayedResponse.cs (hunk 3 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Blobs.Batching/src/DelayedResponse.cs]: Done

**Comment by pakrym** [comment]
Approved.

**Pull request merged**